### PR TITLE
is_available

### PR DESCRIPTION
**Pull request by woodbrettm**
## Tasks Done
- In Games controller def index, added `@game = Game.where(available: true, limit: 10`). The limit is 10 so that when the user scrolls down, this will create an ajax call (not implemented yet) that will call this method again and append the extra 10 games below.
- In game spec, adjusted tests for open spots. Still need to implement destroy game if both seats are open. 
- Added dependent destroy to the pieces table association in the game model
- Added available column in game table, added logic for that.

### CODE REVIEW

File: app/models/game.rb [hunk 1]
```diff
@@ -2,9 +2,20 @@ class Game < ActiveRecord::Base
...
 
-  def open_seat?
-    white_player_id.nil? || black_player_id.nil?
+  after_create :open_seat
+
+  def open_seat
+    white_player_open = white_player_id.nil?
+    black_player_open = black_player_id.nil?
+
+    self.available = if white_player_open || black_player_open
```
Comment (ghost) [inline app/models/game.rb]: Im pretty sure this line could be slimmed down to just

self.available = white_player_open || black_player_open

It just makes it look a little cleaner.

**Comment by mscoutermarsh** [comment]
@bmw95 hey, this is looking good. :smile:

I'm thinking that we'll want users to be able to join a game after it is created. So, for example.
- One player makes a game
- another player comes along, sees it is available & and joins the game

Also.
I have another idea.

Instead of having the `available` column. We are able to determine if a game is available from the existing columns.

So you could do something like this...

``` Ruby
# game.rb
  scope :available, -> { where('white_player_id IS NULL OR black_player_id IS NULL') }
```

Then instead of doing

``` Ruby
Game.where(available: true)

## can do this
Game.available
```

Same functionality, but saves us having to manage that extra column. :smile: 

What do you think?

**Comment by woodbrettm** [comment]
- [x] Get rid of lines 63-65 in gemfile. Factory girl is specified twice currently.

File: app/controllers/games_controller.rb [hunk 2]
```diff
@@ -4,10 +4,15 @@ def new
...
+    @games = Game.open_seats
   end
 
   def create
+    # Note that the creator of the game automatically
+    # becomes the black_player. Due to some issues in
+    # querying, for games index, the model only checks
+    # for games where the white_player_id is nil.
     @game = Game.create(game_params)
     @game.update_attribute(game_params)
```
Comment (mscoutermarsh) [inline app/controllers/games_controller.rb:17]: update_attributes isn't needed. Since those params get set by the previous line.

**Pull request merged**